requests_cache/backends/filesystem.py

```diff
@@ -167,3 +182,247 @@ def _get_extension(extension: Optional[str] = None, serializer=None) -> str:
             name = name.replace(k, v)
         return f'.{name}'
     return '.dat'
+
+
+TEN_MB = 10 * 1024 * 1024
+
+
+class LimitedFileDict(FileDict):
+    """A size-restricted version of the file-dict.
+
+    Args:
+        maximum_cache_bytes: The maximum total size of all files in the cache.
+            This is 10MB by default.
+        block_bytes: The size of a block of data on the file system.
+            The file size will be computed as multiples of this.
+            This is 1 by default.
+        maximum_file_bytes: The maximum size of a single file.
+            By default, this is the same as ``maximum_cache_bytes``.
+
+    File system layout::
+
+        cache_dir/
+            last.int -> int; last file <id>
+            size.int -> int; size of all files added up
+            <key>.<ext>/<id> -> bytes; content
+            ids/<id> -> <key>.<ext>; ordered ids to find oldest fast
+    """
+
+    encoding = 'utf-8'
```

Comment:
Currently there isn't any use case (that I'm aware of) for plaintext encoding other than the default utf-8. Any other encoding/decoding can be done by the serializer, as binary data if necessary.